README.md

```diff
@@ -75,6 +90,60 @@ when it reaches articles that are after the earliest accepted date, it will actu
  'url': 'https://www.iol.co.za/personal-finance/stanlib-may-further-reduce-fund-offering-14717297'}
 ```
 
+### Make a spider
+
+First, add this repository as a remote and make sure your local master is up to date:
+```bash
+git remote add upstream git@github.com:public-people/scrape-news.git
+git fetch upstream
+git checkout master
+git merge upstream/master
+```
+
+Create and check out a branch for the spider you're making:
+```bash
+git checkout -b newssite
+```
+
+Copy a spider from the repository and amend it as necessary. When it's done, ```git add``` it and ```commit``` the change to your working branch.
+
+
+To test individual xpaths you can use the scrapy shell:
+```bash
+scrapy shell "https://www.newssite.co.za/article-url"
+```
+If you go to the same url in your browser and right-click on, say, the title of the article, and select 'Inspect Element (Q)', you'll see something like
+```
+<h1 class="article-title">Title of article</h1>
+```
+highlighted. In the scrapy shell you can then enter something like
+```bash
+>>> response.xpath('//h1/text()').extract_first()
+```
+or
+```bash
+>>> response.css('h1.article-title').extract_first()
```

Comment:
this isn't actually right I don't think? or at least it doesn't extract the text, which the xpath response does.

Reply:
okay so 
```response.css('h1.article-title').css('::text').extract_first()``` 
works, but I'm still not sure why css for classes is preferred?

Reply:
CSS is preferred for classes because you can have multiple classes in the `class` attribute.
Consider
```html
<div class="byline pin-right">John Smith</div>
```
- `xpath('//div[class="byline"]')` won't match because this does exact matching
- `xpath('//div[contains(@class, "byline)]') matches but it also matches `<div class="byline-symbol">` which is not what we want to match so we might not get the data we want.
- `xpath('//div[class="byline pin-right"]')` matches but it's brittle - if they just change where the byline is on the page, this stops matching

`css('div.byline')` is best because it doesn't match `byline-symbol`, and it doesn't care what other classes the element has. Do you reckon we should include this sort of explanation in the README to help people get started?

Reply:
For the css option you can do `css('...').xpath('text()')` but looks like your suggestion of `::text` is equally valid https://doc.scrapy.org/en/latest/topics/selectors.html#id1